### PR TITLE
feat(charting, graphing-solution-set): Charting Item Relative Scoring and Label Characters Limit SC-29795, SC-30682

### PR DESCRIPTION
**Pull request by akshayhiwale**
1. Implemented axis label character limit settings for Charting item
2. Added extra prop on outcome object for scores API for relative scoring
3. Punctuation fix in GSS item.
https://illuminate.atlassian.net/browse/SC-29795
https://illuminate.atlassian.net/browse/SC-30682

### CODE REVIEW

File: packages/charting/configure/src/charting-config.jsx [hunk 1]
```diff
@@ -128,6 +128,7 @@ export class ChartingConfig extends React.Component {
       authorNewCategoryDefaults,
       chartingOptions,
       mathMlOptions = {},
+      charactersLimit,
```
Comment (andreeimiron) [inline packages/charting/configure/src/charting-config.jsx:131]: based on the request on pie-lib side, I suggest to rename this to `labesCharactersLimit`

File: packages/charting/controller/src/index.js [hunk 2]
```diff
@@ -138,10 +151,16 @@ export const getScore = (question, session, env = {}) => {
     }
   }
 
-  return {
+  let score = {
```
Comment (andreeimiron) [inline packages/charting/controller/src/index.js:154]: you can use `const` instead `let`

File: packages/charting/controller/src/index.js [hunk 3]
```diff
@@ -222,10 +241,15 @@ export function model(question, session, env) {
 
 export function outcome(model, session, env) {
   return new Promise((resolve) => {
-    resolve({
-      score: getScore(model, session, env).score,
+    let scoreObject = getScore(model, session, env);
```
Comment (andreeimiron) [inline packages/charting/controller/src/index.js:244]: you can use `const` instead `let`

File: packages/charting/controller/src/index.js [hunk 4]
```diff
@@ -222,10 +241,15 @@ export function model(question, session, env) {
...
+    let result = {
```
Comment (andreeimiron) [inline packages/charting/controller/src/index.js:245]: you can use `const` instead `let`

**Review by andreeimiron — CHANGES_REQUESTED** [review]

File: packages/charting/controller/src/index.js [hunk 5]
```diff
@@ -55,18 +56,25 @@ export const getScore = (question, session, env = {}) => {
...
 
     const scoreForLabelAndValueEditable = (answer, corrAnswer) => {
-      const { value, label } = answer;
-
+      const { value, label, index } = answer;
+      let firstEquality = false;
+      let secondEquality = false;
       maxScore += 2;
 
       if (value === corrAnswer.value) {
```
Comment (andreeapescar) [inline packages/charting/controller/src/index.js:64]: It's a little bit confusing what firstEquality and secondEquality mean.

I assume:
firstEquality - if value is correct
secondEquality - if label is also correct
Is my assumption correct? If yes, can we maybe rename the variables, please?

Maybe we can store those values in variables and refactor where needed.
Something ike
`valueIsCorrect = value === corrAnswer.value`
`labelIsCorrect = checkLabelsEquality(label, corrAnswer.label)`

File: packages/charting/controller/src/index.js [hunk 6]
```diff
@@ -129,6 +139,9 @@ export const getScore = (question, session, env = {}) => {
           result = 0;
           answers[index].correctness.label = 'incorrect';
         }
+        if (value === corrAnswer.value && lowerCase(label) === lowerCase(corrAnswer.label)) {
```
Comment (andreeapescar) [inline packages/charting/controller/src/index.js:142]: Same here. I would store these equalities in a variable and use their value

**Review by andreeapescar — CHANGES_REQUESTED** [review]

File: packages/charting/configure/src/charting-config.jsx (hunk 1 above)
Comment (akshayhiwale) [reply packages/charting/configure/src/charting-config.jsx]: Done

File: packages/charting/controller/src/index.js (hunk 2 above)
Comment (akshayhiwale) [reply packages/charting/controller/src/index.js]: Done

File: packages/charting/controller/src/index.js (hunk 3 above)
Comment (akshayhiwale) [reply packages/charting/controller/src/index.js]: Done

File: packages/charting/controller/src/index.js (hunk 4 above)
Comment (akshayhiwale) [reply packages/charting/controller/src/index.js]: Done

File: packages/charting/controller/src/index.js (hunk 5 above)
Comment (akshayhiwale) [reply packages/charting/controller/src/index.js]: Done

File: packages/charting/controller/src/index.js (hunk 6 above)
Comment (akshayhiwale) [reply packages/charting/controller/src/index.js]: Done

**Pull request merged**